### ISSUE TITLE
Incorrect traces for Trace-ScriptInternal included in 3.0-beta1 output

### ISSUE THREAD

**Issue by fflaten** [nohwnd/Profiler#7]
Using 3.0-beta1, events from `Trace-ScriptInteral.ps1` are included in the output.
Not sure if this is intentional, but if it is then the real issue is that they're missing Duration and Average values.

```
❯ $t.Top50SelfDuration | ft

Percent HitCount Duration         Average          SelfDuration     SelfAverage      Name                     Line Text                                                                  Path
------- -------- --------         -------          ------------     -----------      ----                     ---- ----                                                                  ----
 99,387        1 00:00:01.0139396 00:00:01.0139396 00:00:01.0139396 00:00:01.0139396 t.ps1                       7         Start-Sleep 1                                                 C:\Users\Frode\Git\Pes…
 99,831        3 00:00:01.0184736 00:00:00.3394912 00:00:00.0023327 00:00:00.0007776 <scriptblock>               1 {  & .\Git\Pester\Samples\t.ps1  }                                    <scriptblock>
      0        1 00:00:00         00:00:00         00:00:00.0017084 00:00:00.0017084 Trace-ScriptInternal.ps1  115                 & $ScriptBlock                                        C:\Users\Frode\Documen…
  0,069        1 00:00:00.0007043 00:00:00.0007043 00:00:00.0007043 00:00:00.0007043 t.ps1                      10     Write-Host "SCRIPTBLOCK AFTER - Test = '$test'"                   C:\Users\Frode\Git\Pes…
  0,054        1 00:00:00.0005552 00:00:00.0005552 00:00:00.0005552 00:00:00.0005552 t.ps1                      15 Write-Host "OUTSIDE END - Test = '$test'"                             C:\Users\Frode\Git\Pes…
  0,041        1 00:00:00.0004182 00:00:00.0004182 00:00:00.0004182 00:00:00.0004182 t.ps1                       1 Write-Host "OUTSIDE START - Test = '$test'"                           C:\Users\Frode\Git\Pes…
  0,041        1 00:00:00.0004134 00:00:00.0004134 00:00:00.0004134 00:00:00.0004134 t.ps1                       4     Write-Host "SCRIPTBLOCK BEFORE - Test = '$test'"                  C:\Users\Frode\Git\Pes…
      0        1 00:00:00         00:00:00         00:00:00.0000852 00:00:00.0000852 Trace-ScriptInternal.ps1  113                 [Profiler.Tracer]::Patch($ExecutionContext, $host.UI) C:\Users\Frode\Documen…
      0        1 00:00:00         00:00:00         00:00:00.0000569 00:00:00.0000569 Trace-ScriptInternal.ps1  118                 Set-PSDebug -Trace 0                                  C:\Users\Frode\Documen…
 99,507        1 00:00:01.0151728 00:00:01.0151728 00:00:00.0000341 00:00:00.0000341 t.ps1                      13 & $sb1                                                                C:\Users\Frode\Git\Pes…
 99,391        1 00:00:01.0139893 00:00:01.0139893 00:00:00.0000278 00:00:00.0000278 t.ps1                       9     . $sb2                                                            C:\Users\Frode\Git\Pes…
  0,002        2 00:00:00.0000166 00:00:00.0000083 00:00:00.0000166 00:00:00.0000083 t.ps1                       3 $sb1 = {                                                              C:\Users\Frode\Git\Pes…
  0,001        1 00:00:00.0000150 00:00:00.0000150 00:00:00.0000150 00:00:00.0000150 t.ps1                       8     }                                                                 C:\Users\Frode\Git\Pes…
  0,001        1 00:00:00.0000148 00:00:00.0000148 00:00:00.0000148 00:00:00.0000148 t.ps1                      11 }                                                                     C:\Users\Frode\Git\Pes…
  0,001        2 00:00:00.0000137 00:00:00.0000068 00:00:00.0000137 00:00:00.0000068 t.ps1                       5     $sb2 = {                                                          C:\Users\Frode\Git\Pes…
      0        1 00:00:00.0000037 00:00:00.0000037 00:00:00.0000037 00:00:00.0000037 t.ps1                       6         $test = "hello"                                               C:\Users\Frode\Git\Pes…
```

**Comment by nohwnd** [comment]
It is not intentional unless you run that script, which you probably did not. :) 

For some reason the Set-PSDebug -Trace 1 was not logged and I remember the script not measuring correctly on first run. So I added extra trace-line. Because there would be now 2 one for the enablement (which we ignore) and also another one for the & $ScriptBlock, the second one would be picked up and the whole file loaded, soo all lines from that file now appear in Top50 because that is calculated from file lines. Weird design in hindsight, but it is evolving.

**Comment by fflaten** [comment]
> For some reason the Set-PSDebug -Trace 1 was not logged 

This is too complex for me, but I guess `Set-PSDDebug -Trace 1` enables the calls to `WriteDebugLine` (prior to code execution) in the first place so it make sense that it wouldn't trigger on itself, wouldn't it?

I haven't tried the native profiler yet, so I don't have any tips outside of maybe filtering out lines from `Trace-ScriptInternal`/`Profiler` after all the durations etc. are calculated in the post-processing.

**Comment by nohwnd** [comment]
Fixed by #20
Or at least mitigated.

https://github.com/nohwnd/Profiler/pull/20/files#diff-df712fe76fc70d5d33ab688acda5775a8bab14f8814548d79abbd47af8701aadR218 This is where it is done. The logic now does not load the whole file, it does not touch file at all. It makes processing the log faster, and includes everything that was invoked, including unbound scriptblocks and files that were deleted. The downside is that there is $trace.Files no more, which shown nicely the whole file and performance per line, but I am not sure if that is even practical, because I never needed it. It can be re-enabled by a cmdlet.

**Comment by nohwnd** [comment]
> This is too complex for me, but I guess Set-PSDDebug -Trace 1 enables the calls to WriteDebugLine
Yeah probably I think I was just confused.
